tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedErrorMatcher.cs

```diff
@@ -0,0 +1,135 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format
+{
+    public class UnifiedErrorMatcher
+    {
+        public void AssertErrorsMatch(Exception actualException, BsonDocument expectedError)
+        {
+            expectedError.Elements.Should().NotBeEmpty();
+
+            foreach (var element in expectedError)
+            {
+                switch (element.Name)
+                {
+                    case "isError":
+                        AssertIsError(actualException, element.Value.AsBoolean);
+                        break;
+                    case "isClientError":
+                        AssertIsClientError(actualException, element.Value.AsBoolean);
+                        break;
+                    case "errorContains":
+                        AssertErrorContains(actualException, element.Value.AsString);
+                        break;
+                    case "errorCode":
+                        AssertErrorCode(actualException, element.Value.AsInt32);
+                        break;
+                    case "errorCodeName":
+                        AssertErrorCodeName(actualException, element.Value.AsString);
+                        break;
+                    case "errorLabelsContain":
+                        AssertErrorLabelsContain(actualException, element.Value.AsBsonArray.Select(x => x.AsString));
+                        break;
+                    case "errorLabelsOmit":
+                        AssertErrorLabelsOmit(actualException, element.Value.AsBsonArray.Select(x => x.AsString));
+                        break;
+                    case "expectResult":
+                        AssertExpectResult(actualException, element.Value.AsBsonDocument);
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized error assertion: '{element.Name}'");
+                }
+            }
+        }
+
+        private void AssertErrorCode(Exception actualException, int expectedErrorCode)
+        {
+            actualException.Should().BeOfType<MongoCommandException>();
+            var mongoCommandException = actualException as MongoCommandException;
+
+            mongoCommandException.Code.Should().Be(expectedErrorCode);
```

Comment:
Normally we use this slightly shorter form:

    var mongoCommandException = actualException.Should().BeOfType<MongoCommandException>().Subject;
    mongoCommandException.Code.Should().Be(expectedErrorCode);